### PR TITLE
Add the minotaur NPC

### PR DESCRIPTION
**Pull request by owalk**

### CODE REVIEW

File: game/tiles/entities/characters/NPCs/NPC.py [hunk 1]
```diff
@@ -64,6 +66,10 @@ class NPC(Character):
         Define the follow movement
     thrown_movement(self)
         Throw self from current position
+    charge_movement(self)
+        Charge self from current position
+    rotate_compass+to_target_sprite(self)
```
Comment (Sean-Nishi) [inline game/tiles/entities/characters/NPCs/NPC.py:71]:
```suggestion
    rotate_compass_to_target_sprite(self)
```

File: game/tiles/entities/characters/NPCs/minotaur.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        minotaur_movements_path: str = (
+            settings.CHARACTER_IMAGES + "NPCs/minotaur/minotaur.png"
+        )
+        minotaur_image_rect: pygame.Rect = pygame.Rect(
+            0, 0, self._minotaur_width, self._minotaur_height
+        )
+        # self.setup_import_assets()
+        super().__init__(group, pos, minotaur_movements_path, minotaur_image_rect)
+        self._obstacle_sprites: pygame.sprite.Group = obstacle_sprites
+        self._hitbox: pygame.Rect = self.rect
```
Comment (Sean-Nishi) [inline game/tiles/entities/characters/NPCs/minotaur.py:64]: No offset for minotaur for 3d effect? minotaur width and height are 48 so we can adjust how much smaller the hitbox should be

File: game/tiles/entities/characters/NPCs/minotaur.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        self.radar_set_states(self._player, active_state, passive_state)
+
+        # change state to attack if there is good_entity nearby
+        active_state = self.set_state_charge
+        self.radar_set_state(self._good_sprites, active_state)
+
+        self.move_based_on_state()
+
+    def collision_handler(self) -> None:
+        """Handle collision interactions with environment."""
```
Comment (Sean-Nishi) [inline game/tiles/entities/characters/NPCs/minotaur.py:84]: Need docstring for exceptions raised
https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

File: game/tiles/entities/characters/character.py [hunk 4]
```diff
@@ -85,14 +84,15 @@ def __init__(
...
         self._compass.x: Direction = Direction.right
         self._status: str = "right"
         self._score_controller: ScoreController = ScoreController()
         self._sprite_sheet: SpriteSheet = SpriteSheet()
 
-    def import_assets(self) -> None:
+    def setup_import_assets(self, height, width) -> None:
         """Import and divide the animation image into it's smaller parts.
 
         Called at end of :func:'init()', takes the image with all character animations
```
Comment (Sean-Nishi) [inline game/tiles/entities/characters/character.py:98]: No longer called at end of init()

**Review by Sean-Nishi — CHANGES_REQUESTED** [review]
Only a few comments

File: game/tiles/entities/characters/NPCs/minotaur.py (hunk 2 above)
Comment (owalk) [reply game/tiles/entities/characters/NPCs/minotaur.py:64]: the offset is being moved into the terrain for a bunch of reasons. The short version is that we need precision on the hitboxes for collisions to make sense.

Terrain will be a tile that knows the id of it's terrain, and for obstacle terrain, there will be offset accounted for. I already removed it from every other entity when I was debugging the wall collisions and automated movement stuff.

This is even more apparent when we deal with character-on-character collisions. hiutbox precision within characters makes life make more sense

here's the issue for this https://github.com/Sean-Nishi/Reluctant-Hero/issues/345

File: game/tiles/entities/characters/NPCs/NPC.py [hunk 5]
```diff
@@ -109,12 +117,13 @@ def __init__(
 
         # setting up state machine
         self._states: Enum = Enum(
-            "states", ["Patrol", "Attack", "Flee", "Follow", "Thrown"]
+            "states", ["Patrol", "Attack", "Flee", "Follow", "Thrown", "Charge"]
         )
         self._current_state: Enum = self._states.Patrol
+        self._initial_charge_compass: pygame.Vector2 = pygame.Vector2(0, 0)
```
Comment (Sean-Nishi) [inline game/tiles/entities/characters/NPCs/NPC.py:123]: tile.py has its compass as a pygame.math.Vector2() type. What is the difference between pygame.Vector2 and pygame.math.Vector2?

**Review by Sean-Nishi — APPROVED** [review]

File: game/tiles/entities/characters/NPCs/NPC.py (hunk 5 above)
Comment (owalk) [reply game/tiles/entities/characters/NPCs/NPC.py:123]: it's more convenient syntax, but functionally the exact same thing
image

**Review by Sean-Nishi — APPROVED** [review]

**Pull request merged**